src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2919,6 +2914,23 @@ private BoundExpression BindDelegateCreationExpression(ObjectCreationExpressionS
                     return new BoundDelegateCreationExpression(node, boundLambda, methodOpt: null, isExtensionMethod: false, type: type, hasErrors: !conversion.IsImplicit);
                 }
 
+                else if (analyzedArguments.HasErrors)
+                {
+                    // There is no hope, skip.
+                }
+
+                // 2. A method group
+                else if (argument.Kind == BoundKind.MethodGroup)
```

Comment:
It's possible to keep the original order, if you don't mind checking `analyzedArguments.HasErrors` again and again. ie:
```c
else if ( !argsHaveErrors && methodGroup )
  // ...
else if ( lambda )
  // ...
else if ( !argsHaveErrors && delegate )
  // ...
```
or move the args errors check inside each block making another level of indentation for them.

I don't think an `UnboundLamda` would ever have its `Kind == BoundKind.MethodGroup`, see [the constructors here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/BoundNodes.xml.Generated.cs,4851).